### PR TITLE
Automatically set Electronic Emission Beam Data and System Data length (DIS 6)

### PR DESCRIPTION
**Pull request by fo-ifad**
The "Beam Data Length" field, for Elecromagnetic Emission PDU, in  "IEEE Std 1278.1-1995(R2002)" section  5.3.7.1- > e.1, is defined as :

_**Beam Data Length**. This field shall specify the length of this beam’s data (including track/
jam information) in 32-bit words. The length shall include the Beam Data Length field,
and shall be represented by an 8-bit unsigned integer._

This field is set manually by the user of open-dis-java, but could be set automatically with the formula `((2 * getTrackJamTargets().size()) + 13)

`
The fields are shown in the attached image, and these are grouped in 32-bit words, so that the number of words become apparent.

beamDataLength

### CODE REVIEW

File: src/main/java/edu/nps/moves/dis/ElectronicEmissionBeamData.java [hunk 1]
```diff
@@ -96,14 +96,15 @@ public int getMarshalledSize() {
...
-    public void setBeamDataLength(short pBeamDataLength) {
-        beamDataLength = pBeamDataLength;
-    }
-
     public short getBeamDataLength() {
         return beamDataLength;
     }
 
+    short calculateBeamDataLength() {
+        //bea, length = 13 + 2*track/jamTargets
```
Comment (leif81) [inline src/main/java/edu/nps/moves/dis/ElectronicEmissionBeamData.java:104]:
```suggestion
        //beam length = 13 + 2*track/jamTargets
```

File: src/main/java/edu/nps/moves/dis/ElectronicEmissionSystemData.java [hunk 2]
```diff
@@ -69,14 +69,21 @@ public int getMarshalledSize() {
...
         return systemDataLength;
     }
 
+    private short calculateSystemDataLength() {
+        //systemdata length = 13*beamCount + SUM(beamsCount)2*track/jamTargets
+        short size = 5;
+        for (int i = 0; i < beamDataRecords.size(); i++) {
+            ElectronicEmissionBeamData bd = beamDataRecords.get(i);
+            size += bd.calculateBeamDataLength();
+
```
Comment (leif81) [inline src/main/java/edu/nps/moves/dis/ElectronicEmissionSystemData.java:82]:
```suggestion
```

File: src/main/java/edu/nps/moves/dis/ElectronicEmissionSystemData.java [hunk 3]
```diff
@@ -69,14 +69,20 @@ public int getMarshalledSize() {
...
-        systemDataLength = pSystemDataLength;
-    }
-
     public short getSystemDataLength() {
         return systemDataLength;
     }
 
+    private short calculateSystemDataLength() {
+        //systemdata length = 13*beamCount + SUM(beamsCount)2*track/jamTargets
+        short size = 5;
```
Comment (leif81) [inline src/main/java/edu/nps/moves/dis/ElectronicEmissionSystemData.java:78]: What's the size = 5 represent here?

Comment (fo-ifad) [reply src/main/java/edu/nps/moves/dis/ElectronicEmissionSystemData.java]: The 5 represents 32-bit words for the fields in the pdu shown in  figure  below, and is a part of the **System Data Length.** 

> _**System Data Length.** This field shall specify the length of this emitter system’s data (including
> beam data and its track/jam information) in 32-bit words. The length shall include the System
> Data Length field. This field shall be represented by an 8-bit unsigned integer._
> 

system_data_length

**Comment by leif81** [comment]
Thanks @fo-ifad !

**Pull request merged**